kairon/shared/auth.py

```
@@ -154,6 +156,75 @@ def validate_bot_specific_token(request: Request):
+            raise HTTPException(
+                status_code=status.HTTP_403_FORBIDDEN,
+                detail="Token is not valid for this bot",
+            )
+
+        try:
+            customer = CustomerDetails.objects(bot=bot, sender_id=claims.get("sub")).get()
+        except DoesNotExist:
+            raise HTTPException(
+                status_code=status.HTTP_403_FORBIDDEN,
+                detail="Sender is not registered for this bot",
+            )
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)